enhancements/network/ovn-on-hypershift.md

```diff
@@ -0,0 +1,275 @@
+┌────────┴───────┐        │
+│ ovnkube-master │        │
+│   ovn-northd   │        │
+└────────────────┘        │
+```
+
+**Control-plane:**
+- ovn-kube master: Watches apiserver, creates objects in nbdb
+- ovn-northd: Translates nbdb objects to sbdb flows
+- nbdb, sbdb: Raft-based highly available configuration databases (northbound and southbound)
+
+**Node:**
+- ovn-kube node: Does some node-level configuration
+- ovn-controller: Programs flows from sbdb to Open vSwitch
+- Open vSwitch (OVS): actual packet handling
+
+The control plane components will be deployed as pods on the management cluster. The node components remain on the hosted cluster. The CNO will be made aware of both of these clusters so it can deploy the right compoments in the right place.
+
+The ovnkube-master pod currently runs using host networking. This must be changed to use the cluster network, instead.
+
+The CNO creates a headless service called `ovnkube-db` for the OVN databases. This needs to be updated to have a ClusterIP, since in this updated architecture, we need to point a `Route` at this `Service`. The CNO will also be responsible for creating this `Route` and configuring `ovn-controller` in the hosted cluster to connect to the associated hostname for southbound DB access.
+
+Anywhere the CNO assumes that `openshift-ovn-kubernetes` is the target namespace for OVN components must be updated. That will be the namespace for Node components in the hosted cluster, but not for the components that run in the hosted control plane.
+
+#### OVN Control plane
+
+Deploying the OVN control plane will require some general refactoring, as they no longer will run as HostNetwork, privileged pods. Rather
+
+**ovsdb-server (nbdb, sbdb)**: *should* not require any changes. This depends on Raft working when nodes are identified by DNS (and IPs change).
+
+**OVN-Kubernetes master:** *may* need some changes to perform leader election via the management cluster. Otherwise, it doesn't need to be aware of the management cluster.
+
+#### OVN DBs
+
+The OVN databases (nbdb, sbdb) will continue to use the internal raft-based replication. However, they will be deployed as a StatefulSet instead. Rather than using HostPath volumes, they will use whatever persistent storage volumes are available in the management cluster.
+
+#### OVN Node components
+**ovn-controller**: itself will not require any changes as it appears to already have support for the new functionality required. It must support configuration of the SB (southbound) database connection as a URL instead of an IP address. Further, when using TLS, it must set SNI (Server Name Indication) so that the management cluster is able to route the TLS connection to the correct database instance.
+
+**OVN-Kubernetes node:** should not require any changes.
+
+#### Node-DB connectivity
+By (ab)using the TLS Passthrough feature of OpenShift Routes, we can expose the southbound database directly to the client cluster.
```

Comment:
I think its better to replace "client cluster" with "hosted cluster" because every where else "hosted cluster" is used.